src/captureStartupPerformance.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as vscode from 'vscode';
+import { delay } from 'lodash';
+
+export class CaptureStartupPerformance {
+  constructor() {
+    console.log('Constructor - This is the Capture Startup Performance class!');
+  }
+}
+
+export const captureStartupPerformance = async (): Promise<void> => {
+  console.log('Method - This is the Capture Startup Performance class!');
+
+  try {
+    // Show running notification to the user
+    await vscode.window.withProgress({
+      location: vscode.ProgressLocation.Notification,
+      title: 'SFDX: Capture Startup Performance',
+      cancellable: false
+      }, async (progress) => {
+      progress.report({ message: 'Running...' });
+
+      // open Developer: Startup Performance (open via API - try "perfview.show")
+      await openStartupPerformanceFile();
+      // save startup performance file (TODO: how to do this without popup dialog?)
+      const fileName = await saveCurrentFile();
+      // read startup performance file using fs.readFileSync()
+      const fileContents = await readStartupPerformanceFile(fileName);
+      // parse startup performance file using marked
+      await sleep(2000);
+      const parsedResults = await parseStartupPerformanceFile(fileContents);
+      // save parsed results to a file
+      await copyContentsToFile(parsedResults);
```

Comment:
I was thinking that because they do two separate things, they would be separated.  I called `saveCurrentFile()` twice in the `captureStartupPerformance()` function.

The first time, it is called with the Startup Performance file open in Editor view.  I want to save that file, which can be saved directly into another file without having to explicitly copy the content into another file.

The second time, it is used to save the results of parsing the Startup Performance file.  I need to first call `copyContentsToFile()` to copy the parsed results into a separate new file, then call `saveCurrentFile()` on the new file with the parsed results in it.